addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -449,8 +477,40 @@ def _import_fattura_pa(self, tree, invoice):
         :param invoice: the invoice to update or an empty recordset.
         :returns:       the invoice where the fattura_pa data was imported.
         """
+        buyer_seller_info = self._l10n_it_buyer_seller_info()
+
         company = invoice.company_id
 
+        # There are 2 cases:
+        # - cron:
+        #     * Move direction (incoming / outgoing) flexible (no 'default_move_type')
+        #     * I.e. used for import from tax agency
+        # - "Upload" button (invoices / bills view)
+        #     * Fixed move direction; the button sets the 'default_move_type'
+        default_move_type = self.env.context.get('default_move_type')
+        if default_move_type is None:
+            incoming_possibilities = [True, False]
+        elif default_move_type in invoice.get_purchase_types(include_receipts=True):
+            incoming_possibilities = [True]
+        elif default_move_type in invoice.get_sale_types(include_receipts=True):
+            incoming_possibilities = [False]
+        else:
+            _logger.warning("Cannot handle default_move_type '%s'.", default_move_type)
+            return
+
+        for incoming in incoming_possibilities:
+            company_role, partner_role = ('buyer', 'seller') if incoming else ('seller', 'buyer')
+            company_info = buyer_seller_info[company_role]
+            elements = tree.xpath(company_info['vat_xpath'])
+            if elements and elements[0].text.casefold() in (company.vat or '').casefold():
+                break
+            elements = tree.xpath(company_info['codice_fiscale_xpath'])
+            if elements and elements[0].text.casefold() in (company.l10n_it_codice_fiscale or '').casefold():
+                break
+        else:
+            invoice.message_post(body=_('Company not found (in a fitting role).'))
```

Comment:
```suggestion
            invoice.message_post(body=_('Your company's VAT number and Fiscal Code haven't been found in the buyer and/or seller sections inside the document.'))
```
In the 15.0 version it was like 'Could not determine company (by looking at the VAT and codice fiscale in the buyer and/or seller section).'. I understand that now we have the company for granted, but as you can see there's more information in the old messsage.

This message will stand alone in the chatter after the import on a blank invoice, with no context for the user. A couple more words don't hurt in this case.

Reply:
Thanks your version is much better